en/middleware/microdds.md

```diff
@@ -0,0 +1,74 @@
+
+- [RTPS/DDS Interface: PX4-Fast RTPS(DDS) Bridge](https://docs.px4.io/v1.13/en/middleware/micrortps.html)
+  - [Manually Generate Client and Agent Code](https://docs.px4.io/v1.13/en/middleware/micrortps_manual_code_generation.html)
+- [Fast DDS Installation](https://docs.px4.io/v1.13/en/dev_setup/fast-dds-installation.html)
+:::
+
+
+## Why this approach is of interest?
+
+Initially, micro-RTPS served as the middleware and its consistent improvement evolved into what's known as the microDDS that provides the user a native interface between the flight controller and the mission computer DDS/ROS2 data space. 
+The notable difference between the two is what kind of transport protocol each supports. The newest version supports UDP/TCP/Serial/Custom transport protocol.  
+
+The Micro-XRCE-DDS-Gen has 2 major advantages, i.e. it can be used to generate client code and build px4_msgs along with microdds_client together creating a single library.  
+
+:::tip
+
+:::
+
+## Architectural overview
+
+### microDDS Bridge
+
+[*micro XRCE-DDS*](https://micro-xrce-dds.docs.eprosima.com/en/stable/introduction.html) provides a DDS publisher/subscriber middleware wherein the client is on a low-resource consumption device and the agent is connected to with the DDS global data space.
+
+#### The microdds Client
+The *Client* is the PX4 Autopilot middleware daemon process that runs on the flight controller. The client can either publish or subcribe directly to data topics because the agent itself acts on behalf of the [client](https://micro-xrce-dds.docs.eprosima.com/en/stable/client.html)
+
+#### The microdds Agent
+The *Agent* runs as a daemon process that's defined by what the clients defines it to be and hence it's dynamic. It's a server that acts an an intermediary between the client and the DDS Global Data Space. 
+
+#### microdds Communication
+microDDS has APIs that allow writing time-critical applications and can be used over many transport protocols supporting TCP,UDP over Ethernet Wi-Fi & 6LoWPAN and Bluetooth. The user can also customise this making microDDS transport-agnostic.
+
+## Test
+
+Let’s test! 
+You can check the microdds running using a grep command (ps aux | grep micro)
+
+TERMINAL 1
+
+Start your favorite simulation model, we’re gonna go with the default. 
+
+```sh
+cd ~/PX4_Autopilot #Into your Autopilot folder
+make px4_sitl gazebo 
+```
+Gazebo must open and you can check the micro-dds client running with the following command
+
+```sh
+microdds_client status #Checks client status 
+```
+![Simulation](../../assets/middleware/microdds/simulation.png)
+
+TERMINAL 2
+
+```sh
+cd ~/px4_ros2_ws #Intro your ROS2 directory 
+source /opt/ros/humble/setup.bash
+source install/setup.bash
+ros2 topic list -v #Displays available topic 
+ros2 topic echo fmu/out/vehicle_status #You can check any available topic
+```
+![ROS2 Topic](../../assets/middleware/microdds/topic.png)
```

Comment:
This tutorial stops early. Looking at this more complete one: https://gist.github.com/julianoes/adbf76408663829cd9aed8d14c88fa29#offboard-example-and-px4_msgs

This imports the following:
```
git clone https://github.com/PX4/px4_msgs.git
git clone https://github.com/PX4/px4_ros_com.git src/px4_ros_com
git clone https://github.com/Jaeyoung-Lim/px4-offboard.git src/px4-offboard
````

@Jaeyoung-Lim 
1. I understand that ROS needs to know what ROS nodes are present, and that these are provided by `PX4/px4_msgs` - is that right? 
2. Presumably the definitions change depending on the build. Does that mean we'll take release snapshots of these.
3. If I build locally, where do the px4_msgs get generated? Or to put it another way, if I create a custom build of PX4, how do I expose messages to my ROS setup?
4. These look just like normal uorb topic definitions - how do they get turned into something your script/ROS can use?
5. I thought that `px4_ros_com` was no longer needed? Why does that get imported?

Reply:
Actually, thinking about this, my guess is that  you need `PX4/px4_msgs` if working with XRCE-DDS "alone" in order to tell the toolchain how to build the ROS topics. I assume that the colcon build builds everything, and somewhere in the definitions there is info to tell it to build the ROS2 topics from the messages.

I don't think you need the px4_ros_com at all, ever. If I'm wrong, when is it useful?

So how is microROS different? My guess based on the presentation is that you don't need px4_msgs either. I think in this case the client sends ROS messages by default so there is no need to do any conversion later on. Is that correct? IF not, can you clarify the "usage" difference?

Reply:
`PX4/px4_msgs` is still needed regardless which "agent", bare XRCE-DDS or microROS, you use. Currently it provides the message definitions to ROS2 (you can try running either agents without this package sourced, no error will appear until you actually try to echo a topic, then ROS2 will complain about the missing message definition). I don't know if in the future this will change, like it will be possible to compile the messages directly from the PX4 main repo.

The best part about this new system is that the message definitions on the PX4 side and the ROS2 side are the same. Therefore you only need a copy-paste of all the `.msg` files from PX4 to ROS2 to keep the two in synchronized.

px4_ros_com as you said is no more needed. It contains a few examples of nodes and a library to transform frames between ROS2 and PX4 conventions, but it is no more a dependency.

According to the official repo https://github.com/micro-ROS/micro-ROS-Agent, microROS agent is just a ROS2 node that wraps the micro XRCE-DDS Agent. So the communication is always between micro XRCE-DDS Agent and micro XRCE-DDS Client. The agent then exposes the topics to ROS2 .